workspaces/arborist/lib/override-set.js

```diff
@@ -55,7 +92,8 @@ class OverrideSet {
         return rule
       }
 
-      let spec = npa(`${edge.name}@${edge.spec}`)
+      // We need to use the rawSpec here, because the spec has the overrides applied to it already.
+      let spec = npa(`${edge.name}@${edge.rawSpec}`)
```

Comment:
@AlonNavon Is the `rawSpec` here the original value or the original spec value?
Should an override have something like a `subSpec` to hold the original spec?

Reply:
For the separate PR. Below more unsafe prop access.
```
 if (spec.type === 'git') {
    if (spec.gitRange && semver.intersects(spec.gitRange, rule.keySpec)) {
      return rule
    }
    continue
  }
  if (spec.type === 'range' || spec.type === 'version') {
    if (semver.intersects(spec.fetchSpec, rule.keySpec)) {
      return rule
    }
    continue
  }
   ```
   The property `rule.keySpec` may be undefined so needs guards like with `spec.gitRange`, e.g. `if (spec.gitRange && `.

Reply:
rawSpec is the original value.
overrides.keySpec contains the override value.
spec is the calculated value which applies the override if relevant.